### PR TITLE
p2p: move rlpx into separate package

### PR DESCRIPTION
**Pull request by renaynay**
This PR separates methods related to RLPx transport into a separate `rlpx` package. It also implements a wrapper, `rlpxTransport`,  within package `p2p`. This change was made in order to make it easier to send messages to a node without having to go through the obstacles presented in package `p2p`.

### CODE REVIEW

**Comment by renaynay** [comment]
rlpx ping seems to work with local node

```
{Version:5 Name:Geth/v1.9.21-unstable-53761966-20200902/darwin-amd64/go1.14.1 Caps:[eth/63 eth/64 eth/65] ListenPort:0 ID:0x8de0e7ee434752292d1e64f9a45adc4e8626b698d18473d39b4b02c4b2e79067da6b52a51af9c06b8c69d43e0618208974393a24815e5f3efab1a50de521894b Rest:[]}
```

File: p2p/server.go [hunk 1]
```diff
@@ -943,14 +950,14 @@ func (srv *Server) setupConn(c *conn, flags connFlag, dialDest *enode.Node) erro
...
-	remotePubkey, err := c.doEncHandshake(srv.PrivateKey, dialPubkey)
+	remotePubkey, err := c.doEncHandshake(srv.PrivateKey)
 	if err != nil {
 		srv.log.Trace("Failed RLPx handshake", "addr", c.fd.RemoteAddr(), "conn", c.flags, "err", err)
 		return err
 	}
 	if dialDest != nil {
 		// For dialed connections, check that the remote public key matches.
-		if dialPubkey.X.Cmp(remotePubkey.X) != 0 || dialPubkey.Y.Cmp(remotePubkey.Y) != 0 {
+		if dialPubkey.X.Cmp(remotePubkey.X) != 0 || dialPubkey.Y.Cmp(remotePubkey.Y) != 0 { // TODO move this into rlpx (felix will do later)
```
Comment (renaynay) [inline p2p/server.go:960]: @fjl do you want to do this in this PR or in a separate one?

File: p2p/transport.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+	t.conn.SetReadDeadline(time.Now().Add(frameReadTimeout))
+
+	var (
+		msg Msg
+		err error
+	)
+
+	msg.Code, msg.Size, msg.Payload, err = t.conn.ReadMsg()
+	msg.meterSize = msg.Size
```
Comment (fjl) [inline p2p/transport.go:67]: This is not right. The metered size is supposed to be the compressed size, but here we assign the uncompressed size.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**